tests/v2/validation/rbac/rbacuserroles/cluster_role_test.go

```diff
@@ -0,0 +1,141 @@
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/projects"
+	rbac "github.com/rancher/rancher/tests/framework/extensions/rbac"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RBACAdditionalTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (rb *RBACAdditionalTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *RBACAdditionalTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in rb")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+}
+
+func (rb *RBACAdditionalTestSuite) TestRbacClusterOwnerChecks() {
+	subSession := rb.session.NewSession()
+	defer subSession.Cleanup()
+
+	clusterAdmin, standardUserClient, err := rbac.SetupUser(rb.client, rbac.StandardUser)
+	require.NoError(rb.T(), err)
+
+	additionalUser, additionalUserClient, err := rbac.SetupUser(rb.client, rbac.StandardUser)
+	require.NoError(rb.T(), err)
+
+	rb.T().Logf("Adding user as " + rbac.RoleOwner + " to the downstream cluster.")
+	err = users.AddClusterRoleToUser(rb.client, rb.cluster, clusterAdmin, rbac.RoleOwner, nil)
+	require.NoError(rb.T(), err)
+	standardUserClient, err = standardUserClient.ReLogin()
+	require.NoError(rb.T(), err)
+
+	rb.T().Logf("Validating if " + rbac.RoleOwner + " can add another standard user as a project owner")
+	clusterOwnerProject, err := rbac.CreateProject(standardUserClient, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	err = users.AddProjectMember(standardUserClient, clusterOwnerProject, additionalUser, rbac.RoleProjectOwner, nil)
+	require.NoError(rb.T(), err)
+
+	projectListClusterAdditionalUser, err := projects.ListProject(additionalUserClient, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	assert.Equal(rb.T(), 1, len(projectListClusterAdditionalUser.Data))
+	assert.Equal(rb.T(), clusterOwnerProject.Name, projectListClusterAdditionalUser.Data[0].Name)
+
+	err = users.RemoveProjectMember(standardUserClient, additionalUser)
+	require.NoError(rb.T(), err)
```

Comment:
Assertion is missing?

Reply:
as this is a refactor, I haven't changed the existing logic for any of these tests but updated the formatting but sure I can add an assertion here.

Reply:
Good to have the assertion : the removed user do not have access to the project/cluster anymoe.